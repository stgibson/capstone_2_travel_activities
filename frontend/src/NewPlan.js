import React from "react";

const NewPlan = () => {
  return <div>New Plan Page</div>;
};

export default NewPlan;