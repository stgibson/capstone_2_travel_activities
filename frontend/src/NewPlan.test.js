import React from "react";
import { MemoryRouter } from "react-router-dom";
import { createStore } from "redux";
import { Provider } from "react-redux";
import { render } from "@testing-library/react";
import NewPlan from "./NewPlan";
import rootReducer from "./reducers/root";

const store = createStore(rootReducer);

it("renders without crashing", () => {
  render(
    <Provider store={ store }>
      <MemoryRouter><NewPlan /></MemoryRouter>
    </Provider>
  );
});

it("matches snapshot", () => {
  const { asFragment } = render(
    <Provider store={ store }>
      <MemoryRouter><NewPlan /></MemoryRouter>
    </Provider>
  );

  expect(asFragment()).toMatchSnapshot();
});
