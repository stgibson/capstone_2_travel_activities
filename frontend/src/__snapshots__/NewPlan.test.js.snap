// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <h2>
    Create a New Travel Plan
  </h2>
  <form
    class=""
  >
    <div
      class="form-group"
    >
      <label
        class="form-label"
        for="name"
      >
        Travel plan name:
      </label>
      <input
        class="form-control"
        id="name"
        type="text"
        value=""
      />
    </div>
    <div
      class="form-group"
    >
      <label
        class="form-label"
        for="numOfDays"
      >
        Number of days in travel plan:
      </label>
      <input
        class="form-control"
        id="numOfDays"
        type="text"
        value=""
      />
    </div>
  </form>
  <button
    class="btn btn-primary"
    type="button"
  >
    Create Plan
  </button>
</DocumentFragment>
`;
